### PR TITLE
[CST] - Evidence request alert changes on status tab

### PR DESCRIPTION
**Pull request by pmclaren19**
## Summary

**Code Changes:**
- Renamed DueDate.jsx to DueDateOld.jsx and then updated DueDate.jsx so that it had the new design
- Updated DocumentRequestPageContent.jsx (evss) to use DueDateOld.jsx instead of DueDate.jsx
- Updated DocumentRequestPage.jsx to use DueDateOld.jsx instead of DueDate.jsx
- Added changes to claims-status.scss for the new DueDate.jsx component

- Created FilesNeededOld.jsx so that there was a component that we could toggle on 
- Updated RequestedFilesInfo.jsx so that it used the component FilesNeededOld.jsx instead of harding the html
- Created FilesNeeded.jsx with the new design for the primary alert and so that we could toggle on it

- Created WhatWeNeedToDo.jsx to display on the Status Tab and so that we could toggle on it
- Updated helpers.js to have the new functions getTrackedItems() and getFilesNeeded() that are used on WhatWeNeedToDo.jsx
- Modified helpers.js function truncateDescription() so that the maxLength was a prop, defaulted to 120
- Updated ClaimStatusPageContent.jsx (evss) so that there was logic that used the toggle `cstUseClaimDetailsV2` to toggle between WhatWeNeedToDo.jsx and NeedFilesFromYou.jsx
- Updated ClaimStatusPage.jsxso that there was logic that used the toggle `cstUseClaimDetailsV2` to toggle between WhatWeNeedToDo.jsx and NeedFilesFromYou.jsx

**Added/Modified the following Tests**
- src/applications/claims-status/tests/components/ClaimStatusPage.unit.spec.jsx
- src/applications/claims-status/tests/components/DocumentRequestPage.unit.spec.jsx
- src/applications/claims-status/tests/components/DueDate.unit.spec.jsx
- src/applications/claims-status/tests/components/DueDateOld.unit.spec.jsx
- src/applications/claims-status/tests/components/FilesNeeded.unit.spec.jsx
- src/applications/claims-status/tests/components/FilesNeededOld.unit.spec.jsx
- src/applications/claims-status/tests/components/RequestedFilesInfo.unit.spec.jsx
- src/applications/claims-status/tests/components/WhatYouNeedToDo.unit.spec.jsx
- src/applications/claims-status/tests/components/evss/ClaimStatusPageContent.unit.spec.jsx
- src/applications/claims-status/tests/utils/helpers.unit.spec.js

## Figma Designs

[Mocks in Figma](https://www.figma.com/file/F8U4wddaFouUPVd4mGBMDI/CST---Evidence-Submission---Spec?type=design&node-id=0-809&mode=design&t=vr7AAdcV08s2Uxmp-0)
[Figma new second alert design](https://www.figma.com/file/F8U4wddaFouUPVd4mGBMDI/CST---Evidence-Submission---Spec?type=design&node-id=0-1509&mode=design&t=KEAoDGS1gld7AhAm-0)

## Acceptance Criteria

- [x] There are clear alerts on the status page identifying evidence requests for the veteran
- [x] The header for the "what you need to do" section has to clearly differentiate this section for the user from other parts of the page.
- [x] Different evidence requests can be distinguished from one another and contain enough detail to provide a clear action item for the veteran
- [x] Contains a header that says 'What you need to do'
- [x] First Party Alerts show up under the Status tab
- [x] When a user selects 'view details' on a First Party Alert and they provided the requested information, the alerts for those requests go away (speak with @jerekshoe on if this currently happens, looks like it doesnt in Dev env)
- [x] Body text descriptions cannot be insanely long (apply existing logic from old cards if it's the best way to accomplish this)
- [x] Content elements/fields (such as dates) will conform to the previous alert designs and provide the same information to veterans
- [x] When there are no First Party Alerts, an empty state message appears under the header that says 'There's nothing that we need from you right now. We'll let you know when there's an update.'
- [x] New functionality is behind the feature flag `cst_use_claim_details_v2`

## Related issue(s)

- _Link to ticket created in va.gov-team repo_
department-of-veterans-affairs/va.gov-team#69450

## Testing done

Please view https://github.com/department-of-veterans-affairs/va.gov-team/issues/69450

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| EVSS  | Screenshot 2024-01-19 at 10 19 56 AM | Screenshot 2024-01-18 at 3 18 03 PM |
| LIGHTHOUSE | Screenshot 2024-01-19 at 10 19 41 AM | Screenshot 2024-01-22 at 1 00 19 PM |
| EVSS  | Screenshot 2024-01-22 at 12 32 53 PM|
| LIGHTHOUSE |Screenshot 2024-01-19 at 3 03 45 PM|

## What areas of the site does it impact?

*(Describe what parts of the site are impacted **if** code touched other areas)*

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [x] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [x] Browser console contains no warnings or errors.

### Authentication

- [x] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

**Comment by pmclaren19** [comment]
@skylerschain please review the provided screenshots when you have a chance and let me know if you have any questions/concerns with the design I implemented!

File: src/applications/claims-status/components/DueDateOld.jsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import moment from 'moment';
+
+export default function DueDateOld({ date }) {
+  const now = moment();
+  const dueDate = moment(date);
+  const className = dueDate.isBefore(now) ? 'past-due' : 'due-file';
+  return (
+    <div className="tracked-item-due">
+      <strong className={className}>
+        <i className="fa fa-exclamation-triangle past-due-icon" /> Needed from
```
[the remark this answers is not in the document]
Comment (pmclaren19) [reply src/applications/claims-status/components/DueDateOld.jsx:12]: This is an already existing icon that will eventually be deprecated.

File: src/applications/claims-status/tests/components/evss/ClaimStatusPageContent.unit.spec.jsx [hunk 2]
```diff
@@ -7,24 +7,65 @@ import { $ } from '@department-of-veterans-affairs/platform-forms-system/ui';
...
-  context('when cstUseClaimDetailsV2 feature flag enabled', () => {
-    const claim = {
-      attributes: {
-        phase: 2,
-        decisionLetterSent: true,
-        open: true,
+  const getStore = (cstUseClaimDetailsV2Enabled = true) =>
+    createStore(() => ({
+      featureToggles: {
+        // eslint-disable-next-line camelcase
```
[the remark this answers is not in the document]
Comment (pmclaren19) [reply src/applications/claims-status/tests/components/evss/ClaimStatusPageContent.unit.spec.jsx:13]: Needed for testing the toggle can ignore.

**Comment by skylerschain** [comment]
Reviewed with @pmclaren19 and LGTM 👍 !

**Comment by pmclaren19** [comment]
Met with Skyler and we added a few css changes to the alert design to make it more consistent with the secondary alert. Alert now looks like this 
Screenshot 2024-01-24 at 12 01 50 PM

File: src/applications/claims-status/utils/helpers.js [hunk 3]
```diff
@@ -79,6 +79,32 @@ export const getTrackedItemDate = item => {
 };
 // END lighthouse_migration
 
+export function getTrackedItems(claim, useLighthouse = true) {
+  // claimAttributes are different between lighthouse and evss
+  // Therefore we have to filter them differntly
+  // const { trackedItems } = claim.attributes;
```
Comment (samcoforma) [inline src/applications/claims-status/utils/helpers.js:85]: Can we remove this comment?

**Review by samcoforma — DISMISSED** [review]
One nit, but otherwise LGTM

File: src/applications/claims-status/utils/helpers.js (hunk 3 above)
Comment (pmclaren19) [reply src/applications/claims-status/utils/helpers.js]: updated thanks!

**Review by samcoforma — APPROVED** [review]

**Pull request merged**